pkg/ccl/sqlproxyccl/interceptor/frontend_interceptor.go

```diff
@@ -15,20 +15,16 @@ import (
 	"github.com/jackc/pgproto3/v2"
 )
 
-// FrontendInterceptor is a client interceptor for the Postgres frontend
-// protocol.
+// FrontendInterceptor is a client interceptor for the Postgres frontend protocol.
 type FrontendInterceptor pgInterceptor
 
-// NewFrontendInterceptor creates a FrontendInterceptor. bufSize must be at
-// least the size of a pgwire message header.
-func NewFrontendInterceptor(
-	src io.Reader, dst io.Writer, bufSize int,
-) (*FrontendInterceptor, error) {
-	pgi, err := newPgInterceptor(src, dst, bufSize)
-	if err != nil {
-		return nil, err
-	}
-	return (*FrontendInterceptor)(pgi), nil
+// NewFrontendInterceptor creates a FrontendInterceptor. If bufSize is smaller
+// than 5 bytes, the defaults (8K) will be used.
+//
+// NOTE: For future improvement, we can use the options pattern here if there's
```

Comment:
nit: I recommend removing the bufSize argument. We do not need it and it would be a backwards compatible change to add it as a varg option in the future.